### PR TITLE
Ensure that Jest testRegex catches all the specs

### PR DESCRIPTION
**Pull request by vojtechszocs**
Fixes CONSOLE-1658

The change in Jest [`testRegex`](https://jestjs.io/docs/en/configuration.html#testregex-string-array-string) made in commit b4bc8c9 caused some specs not to be executed.

Files under `__tests__` are either actual tests or test helpers (utils, mocks, etc). Actual tests should have `spec` in their filename. We want to co-locate test helpers with actual tests, but we also want Jest to not match those helpers.

```sh
$ NON_SPEC_FILES=`find . -type f -regex '.*/__tests__/.*' | grep -vF 'node_modules' | grep -vF '.spec.'`
$ grep -l 'describe' $NON_SPEC_FILES 
./__tests__/components/utils/datetime.ts
./__tests__/public/co-fetch.js
./__tests__/resource.js
./__tests__/selector-requirement.js
./__tests__/selector.js
./__tests__/container.ts
./__tests__/probe.ts
./__tests__/units.js
```

Above ^^ files have been git-mv'ed to contain `spec` in their filename.

`frontend/__tests__/units.spec.js` is still failing, so using `xdescribe` as a temporary fix. I've talked with @rawagner and he'll take a closer look on that.

cc @jelkosz

### CODE REVIEW

**Comment by spadgett** [comment]
We'll see if the tests still pass... :/

**Comment by vojtechszocs** [comment]
There are some test failures, I'll fix them.

File: frontend/package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "transformIgnorePatterns": [
       "<rootDir>/node_modules/(?!(lodash-es|@console|@novnc|@spice-project)/.*)"
     ],
-    "testRegex": "/__tests__/.*\\.spec\\.(ts|tsx|js|jsx)$",
+    "testRegex": "/__tests__/.*\\.(ts|tsx|js|jsx)$",
```
Comment (christianvogt) [inline frontend/package.json]: This can't be right. We need `spec` in there because we put non-test files inside the `__tests__` dir as support utils or data.

Comment (vojtechszocs) [reply frontend/package.json]: Exactly, we need this for all the monorepo tests, otherwise it blows up on test helpers (and rightly so).

```sh
$ yarn test packages
..
packages/dev-console/src/components/pipelines/__tests__/pipeline-mock.ts
..
Your test suite must contain at least one test.
```

We could use Jest [projects](https://jestjs.io/docs/en/configuration.html#projects-array-string-projectconfig) feature, or just git-mv the 8 files in `frontend/__tests__` to correctly follow the `spec` convention.

Comment (christianvogt) [reply frontend/package.json]: @vojtechszocs just make `testRegex` an array. One for `packages` and one for everything else.

Comment (christianvogt) [reply frontend/package.json]: @vojtechszocs actually just move the files to the correct format and be done with it. It's where we want to go anyways.

**Comment by vojtechszocs** [comment]
PR updated, git-mv'ed above mentioned files to ensure they are matched by Jest.

File: frontend/__tests__/units.spec.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import * as _ from 'lodash-es';
 
 import { units, validate, convertToBaseValue } from '../public/components/utils/units';
 
-describe('units', () => {
+xdescribe('units', () => {
```
Comment (spadgett) [inline frontend/__tests__/units.spec.js]: We should really try to fix these tests. How bad are the failures?

Comment (vojtechszocs) [reply frontend/__tests__/units.spec.js]: @rawagner has fixed the tests :tada: 

He said the problem originates from commit https://github.com/openshift/console/commit/b14d83ba42bb1f749d515cddc697ffc526c98689 (change in `round` function).

**Comment by vojtechszocs** [comment]
@christianvogt @alecmerdler Tests are fixed now, can you please take a look if you have some time?

**Comment by christianvogt** [comment]
Seems ok, but unsure of the history and ux expectation for the unit formatting.

File: frontend/public/components/utils/units.js [hunk 3]
```diff
@@ -175,16 +184,21 @@ export const humanizeCpuCores = v => {
...
   };
 };
-export const humanizePercentage = value => ({
-  string: formatPercentage(value/100),
-  unit: '%',
-  value,
-});
+export const humanizePercentage = value => {
+  if (!isFinite(value)) {
+    value = 0;
```
Comment (spadgett) [inline frontend/public/components/utils/units.js]: I would expect this to return the empty string to differentiate between 0 and no value

File: frontend/public/components/utils/units.js [hunk 4]
```diff
@@ -134,6 +133,14 @@ const round = units.round = (value, options) => {
   return Intl.NumberFormat(locales, rest).format(value);
 };
 
+const round = units.round = (value, fractionDigits) => {
+  if (!isFinite(value)) {
+    return 0;
```
Comment (spadgett) [inline frontend/public/components/utils/units.js]: Again, not sure treating non finite values as 0 is correct

File: frontend/public/components/utils/units.js [hunk 5]
```diff
@@ -122,10 +122,9 @@ const getDefaultFractionDigits = value => {
   return 1;
 };
 
-const round = units.round = (value, options) => {
-  const fractionDigits = getDefaultFractionDigits(value);
+const formatValue = (value, options) => {
   const {locales, ...rest} = _.defaults(options, {
-    maximumFractionDigits: fractionDigits,
+    maximumFractionDigits: 20,
```
Comment (spadgett) [inline frontend/public/components/utils/units.js]: What is the reason for this change? 20 seems like a big value, but maybe I'm misunderstanding this option.

Comment (rawagner) [reply frontend/public/components/utils/units.js]: fraction digits are determined by `round` function https://github.com/openshift/console/blob/21c5ba2b87b0fbf97e4f7034e27b5dd2fe0be5d6/frontend/public/components/utils/units.js#L140 and `formatValue` function is used only for formatting string part - I didnt want to change value here so I set `maximumFractionDigits` to maximum that is allowed

Comment (spadgett) [reply frontend/public/components/utils/units.js]: Do we need `round` at all if we can use `maximumFractionDigits` for formatting here?

This should at least have a comment explaining.

File: frontend/public/components/utils/units.js (hunk 4 above)
Comment (rawagner) [reply frontend/public/components/utils/units.js]: This is the behavior which we had before tests started to break.
Humanize functions return object which is supposed to look like (with added types)
```
value: number;
unit: string;
string: string;
```
`round` function is operating on `value` part and result should be number.

File: frontend/public/components/utils/units.js (hunk 5 above)
Comment (rawagner) [reply frontend/public/components/utils/units.js]: humanize functions return object
```
value: number;
unit: string;
string: string;
```
at first we work on `value` -> it gets rounded (result is number type) and then the rounded value is passed to `formatValue` which is using `Intl.NumberFormat.format` and the result is string type.

I we use only `Intl.NumberFormat.format` then we would need to parse the resulting string back to number which seems a more clumsy to me.

I will add a comment explaining this if we decide to go with current approach

Comment (spadgett) [reply frontend/public/components/utils/units.js]: So I'm still unclear why we need to change this implementation here. Is this fixing a bug, changing behavior, or just refactoring? Can you give an example of the before / after with this change?

Based on your description, I think the existing API is a bit confusing. I would expect

* A function that takes a string value with units and returns an object like `{value, unit}`
* A function that takes the base, unitless value, picks the best unit, and formats it using `Intl.NumberFormat` and specifying significant digits
* A function that takes `{value, unit}` and formats it using `Intl.NumberFormat` and specifying significant digits

I'm not proposing we make these changes in this PR, but we might look at refactoring this down the road.

Comment (rawagner) [reply frontend/public/components/utils/units.js]: If we keep the `fractionDigits` in `Intl.NumberFormat` config and call `humanize` with `useRound=false` we will end up with `value` not rounded, but `string` rounded. Maybe that is not an issue, `useRound=false` isnt even used anywhere in code right now and when we refactor things in the future we will either way end up with something different.

Comment (spadgett) [reply frontend/public/components/utils/units.js]: > If we keep the fractionDigits in Intl.NumberFormat config and call humanize with useRound=false we will end up with value not rounded, but string rounded. 

This is exactly what I would expect :) I think it's a better API. Is this what the code was doing before?

Comment (rawagner) [reply frontend/public/components/utils/units.js]: Ok :) Reverted the change. Before we started using `Intl.NumberFormat` both `value` and `string` was the same (`string` was just `value` + `space` + `unit`), after `Intl.NumberFormat` was introduced the things got different/tests started to fail so I wasnt sure if this change was on purpose or not. But I agree that it makes sense how to keep the logic as is.

Comment (spadgett) [reply frontend/public/components/utils/units.js]: OK, I see the complaint now. It's the the string value is rounded when useRound = false. Sorry I misunderstood.

I think the fix is to always round the humanized value and just remove `useRound` if it's not used anywhere today. But that can be a follow on.

**Comment by spadgett** [comment]
Looks like we have one test failure:

```
FAIL __tests__/units.spec.js (91.846s)
  ● units › should humanize numeric values › 0.1234 into 0.1234 
```

**Comment by spadgett** [comment]
/lgtm cancel
as we have a test failure

**Comment by rawagner** [comment]
sorry about that, i forgot to push latest changes, should be fine now.
